scripts/quiz-score-sync/README.md

```diff
@@ -0,0 +1,72 @@
+# CS169A Quiz Score Upload & Report Script
+
+A script to upload CS169A PrairieLearn quiz scores to bCourses and generate insightful reports for course administration.
+
+## Features
+- **Gradebook Update**: Merges the existing bCourses gradebook with PrairieLearn quiz scores using a left join to create an updated gradebook.
+- **Quiz Participation Report**: Generates a report listing students who haven't taken quizzes.
+- **Anomaly Detection**: Identifies inconsistencies, such as students present in quiz scores but missing from bCourses due to mismatched email addresses.
+- **Separate Logging**: Logs are saved to a `script.log` file, ensuring that reports remain clean and free from log entries.
+- **Automated Testing**: Includes a test script to verify the correctness of report generation and gradebook updates without affecting your working directories.
+
+## Requirements
+- **Python 3**
+- **Pandas** library
+
+## Installation
+
+1. **Install Python 3**
+
+2. **Install Pandas**
+   ```sh
+   pip install pandas
+   ```
+
+## Usage
+
+### 1. **Obtain Required CSV Files**
+   - **Quiz Scores CSV from PrairieLearn**: Download the quiz score CSV file from PrairieLearn.
+![img.png](img/img.png)
+   - **Gradebook CSV from bCourses**: Download the latest gradebook CSV file from bCourses.
+![img_1.png](img/img_1.png)
+### 2. **Run the Script**
+   ```sh
+   python3 update_gradebook_quiz_score.py <prairielearn_quiz_score_csv> <bcourses_gradebook_csv> --output-dir <output_directory>
+   ```
+   **Example:**
+   ```sh
+   python3 update_gradebook_quiz_score.py CS_169_Fall-2024_Q3_scores.csv 2024-10-26T1436_Grades-COMPSCI_169A_FA24.csv --output-dir ./output
+   ```
+
+   **Parameters:**
+   - `<prairielearn_quiz_score_csv>`: Path to the PrairieLearn quiz scores CSV file.
+   - `<bcourses_gradebook_csv>`: Path to the bCourses gradebook CSV file.
+   - `--output-dir <output_directory>`: (Optional) Directory to save the updated gradebook and reports. Defaults to `output` if not specified.
+
+### 3. **Output Files**
+   - **[updated_gradebook_with_quiz_score.csv](tests%2Fdata%2Fexpected_output_sanitized%2Fupdated_gradebook_with_quiz_score.csv)**: The merged and updated gradebook containing quiz scores.
+   - **[reports.txt](tests%2Fdata%2Fexpected_output_sanitized%2Freports.txt)**: A detailed report listing students who haven't taken quizzes and any anomalies detected.
```

Comment:
I don't think these links will work since they're URL encoded.

Reply:
> I don't think these links will work since they're URL encoded.

Looks like they are working. I used a relative URL path, and it seems GitHub automatically determines the absolute path for us. For example, when you go to the [rendered README file](https://github.com/saasbook/courseware/blob/566cf0973865685413a39536fcd7cb127d47f762/scripts/quiz-score-sync/README.md#3-output-files) and click the link, GitHub successfully redirects us to the file stored in the repository.